.github/workflows/python.yml

```diff
@@ -138,6 +138,17 @@ jobs:
         continue-on-error: true
         run: archery docker push ${{ matrix.image }}
 
+      - name: Type check with mypy and pyright
+        run: |-
+            python -m pip install mypy pyright ty griffe libcst pytest hypothesis fsspec scipy-stubs pandas-stubs types-python-dateutil types-psutil types-requests griffe libcst sphinx types-cffi
+            pip install -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple pyarrow
+            cd python
+            mypy
+            pyright
+            ty check
+            cd ..
+            python ./dev/update_stub_docstrings.py -f ./python/pyarrow-stubs
```

Comment:
Is the `update_stub_docstrings.py` something:
- devs should run before pushing or via commit hook
- CI should run on every PR
- something we should run when building sdist/wheels
- part of the release process

It's not clear to me.